### ISSUE TITLE
Fine-tune on a custom dataset

### ISSUE THREAD

**Issue by mJones00** [brjathu/LART#12]
@brjathu Thank you for sharing this amazing work.
I have a question regarding finetuning on a custom dataset. I have run the demo on some videos to get the results_temporal_fast pkl files. I modified them to use my gt labels instead of the 80 labels of ava dataset. 

I noticed that the demo provides ways for predicting both ava labels and kinetics labels. I'm not sure if I should replace the ava part or the kinetic part with my gt. The code seems to use pseudo labels for ava label space as well, so the dimension of the pseudo labels is not the same as my gt labels. Should I get rid of the pseudo label part since I have gt annotation for each frame ?

My other question is how can I use the pretrained model that you provide to finetune it on my dataset of smaller number of classes ?
Thanks

**Comment by brjathu** [comment]
Hi, @mJones00, Thanks for your interest in our work!
That is great! If your task is multi-class classification you could replace Kinetics labels, if it is multi-label classification then you can replace AVA labels. you can change the number of classes in the config here: https://github.com/brjathu/LART/blob/829aaae9dfd332706f92c463d1c55a26c46ac325/configs/lart.yaml#L168C9-L168C33. Since you have the ground truth data, yes you can get rid of the pseudo labels. Also please double check, that the `has_gt` in the data loader is properly configured, and if the sample has gt, `has_gt` should be equal to 2 (0 for no labels and 1 for pseudo labels.)

For the second part, can you only load the transformer weights remove the ava-head, and attach your own classification head?

I hope this helps, Please feel free to ask if you have any further questions :)

**Comment by brjathu** [comment]
Hi, oh that is great, glad you were able to train on your own data. 
It seems the model is training, but since you were only training the head, it might not perform well. 
The code has layer decay implemented, which would be helpful when finetuning on small data, so I would recommend, enabling layer decay (https://github.com/brjathu/LART/blob/829aaae9dfd332706f92c463d1c55a26c46ac325/configs/lart.yaml#L158) and finetune the whole model.

**Comment by brjathu** [comment]
closing due to inactivity, please reopen if you have any questions.

**Comment by vtn98** [comment]
Hi @mJones00,
Could you provide more details for training on a custom dataset, please?
Thank you!
